auth/src/main/java/org/openedx/auth/presentation/signin/compose/SignInView.kt

```diff
@@ -185,6 +187,20 @@ internal fun LoginScreen(
                             state,
                             onEvent,
                         )
+                        state.agreement?.let {
+                            Spacer(modifier = Modifier.height(24.dp))
+                            val linkedText =
+                                TextConverter.htmlTextToLinkedText(state.agreement.label)
+                            HyperlinkText(
+                                modifier = Modifier.testTag("txt_${state.agreement.name}"),
+                                fullText = linkedText.text,
+                                hyperLinks = linkedText.links,
+                                linkTextColor = MaterialTheme.appColors.primary,
+                                action = { link ->
```

Comment:
we can check the tapped linked here and only passed the required data.
```
action = { link ->
    linkedText.links.filterValues { value -> value == link }
    .firstNotNullOf { entry ->
    onEvent(AuthEvent.OpenLink(entry.key, entry.value))
  }
},

Reply:
we shouldn't paste logic in the UI level. So that's why all my logic is executed in the view model